src/filter/segfilter.cpp

```diff
@@ -1,6 +1,21 @@
 #include "segfilter.hpp"
 #include "../Exporter/exportimage.hpp"  // Include ExportImage header
 
+SegFilter::SegFilter(const std::vector<std::string>& modelPaths, ThreadLogger* logger)
+        : logger(logger) {
+    for (const auto& path : modelPaths) {
+        IModel* model = new TensorRTModel();
```

Comment:
I don't think this should be in the instance constructor. 
maybe it could be done in a way that the models can be requested per idx so it can be iterated in the pipeline.